api/sessions_api.go

```diff
@@ -306,6 +306,68 @@ sessionLoop:
 				break sessionLoop
 			}
 
+		case mc.CodeRematchCall:
+			// 1. See if the game still exists
+			game, err := s.GameManager.FindGame(s.GameUuid)
+			if err != nil {
+				break sessionLoop
+			}
+
+			if game.IsRematchAlreadyCalled() {
+				continue sessionLoop
+			}
+
+			game.CallRematch()
+
+			// 2. Find the other player
+			otherPlayer := game.HostPlayer
+			if s.Player.IsHost {
+				otherPlayer = game.JoinPlayer
+			}
```

Comment:
I feel like we have this logic duplicated many times in the codebase, can we make a function out of it?

Reply:
something like this:
```go
func getOtherPlayer(player, game) Player {
	if player.IsHost {
		return game.joinPlayer
	}
	return game.hostPlayer
}

otherPlayer := getOtherPlayer(s.Player, game)

Reply:
on second thought, the `getOtherPlayer` function could be defined on `session` object. so it will look like this:
```go
func getOtherPlayer(game) Player {
    if s.player.IsHost {
        return game.joinPlayer
    }
    return game.hostPlayer
}

otherPlayer := s.getOtherPlayer(game)
```

Reply:
I think I use this in handlers too.
You could argue that players are managed by each game. So, that method could be on game 

```go
func (g *Game) getOtherPlayer(player *Player) *Player {
    if player.IsHost {
        return g.JoinPlayer
    }
    return h.HostPlayer
}
```